pkg/bucket/bucket_controller.go

```diff
@@ -100,15 +113,18 @@ func (b *BucketListener) Add(ctx context.Context, inputBucket *v1alpha1.Bucket)
 		bucketID = bucket.Spec.ExistingBucketID
 		if bucket.Spec.Parameters == nil {
 			bucketClass, err := b.bucketClasses().Get(ctx, bucket.Spec.BucketClassName, metav1.GetOptions{})
-			if err != nil {
+			if kubeerrors.IsNotFound(err) {
+				b.recordEvent(inputBucket, v1.EventTypeWarning, "InvalidBucketClass", "BucketClass provided in the BucketClaim does not exist")
```

Comment:
This event type is also shared here: https://github.com/kubernetes-sigs/container-object-storage-interface-controller/pull/101#discussion_r1396372665

It seems to me that it might be best to include the defined event types with the COSI API repo and reference them here.